### PR TITLE
Replace SourceLookup invocations with Source

### PR DESCRIPTION
**Pull request by romseygeek**
Removes a few more uses of SourceLookup in favour of plain 
Source.  This also allows us to make the MapSourceProvider
implementation within SourceLookup private.

### CODE REVIEW

**Comment by romseygeek** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java [hunk 1]
```diff
@@ -272,12 +273,12 @@ private static Fields generateTermVectors(
         }
         if (source != null) {
             MappingLookup mappingLookup = indexShard.mapperService().mappingLookup();
-            SourceLookup sourceLookup = new SourceLookup(new SourceLookup.MapSourceProvider(source));
+            Source s = Source.fromMap(source, XContentType.JSON);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java:276]: This is so much less confusing! It's not a lookup any more!

**Review by nik9000 — APPROVED** [review]
This is lovely!

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Pull request merged**